tools/riscv/README_RISCV.md

```diff
@@ -0,0 +1,59 @@
+# Building RISC-V Target with Cmake #
+
+## Prerequisite: Build RISC-V Clang Toolchain and QEMU ##
+
+If you don't have prebuilt clang and riscv64 qemu, you can refer to the [script](https://github.com/sifive/prepare-riscv-toolchain-qemu/blob/main/prepare_riscv_toolchain_qemu.sh) to get the source. Copy the script to the zlib-ng root directory, and run it to download the source and build them. Modify the content according to your conditions (e.g., toolchain version).
+
+```
+./prepare_riscv_toolchain_qemu.sh
+```
+
+After running script, clang & qemu are built in `build-toolchain-qemu/riscv-clang/` & `build-toolchain-qemu/riscv-qemu/`.
+
+`build-toolchain-qemu/riscv-clang/` is your `TOOLCHAIN_PATH`.
+`build-toolchain-qemu/riscv-qemu/bin/qemu-riscv64` is your `QEMU_PATH`.
+
+## Cross-Compile for RISC-V Target ##
+
+```
+cmake -G Ninja -B ./build-riscv \
+  -D CMAKE_TOOLCHAIN_FILE=./cmake/toolchain-riscv.cmake \
+  -D CMAKE_INSTALL_PREFIX=./build-riscv/install \
+  -D TOOLCHAIN_PATH={TOOLCHAIN_PATH} \
+  -D QEMU_PATH={QEMU_PATH} \
+  .
+
+cmake --build ./build-riscv --target install
+```
+
+## Enable RISC-V Extensions at Runtime ##
```

Comment:
I apologize for using a misleading name. The feature is designed to get risc-v cpu info, because we can't use hwcap to detect risc-v extensions at runtime in linux. We may find better solution in the future.

Reply:
Enabling auto-vectorization in compiler and run-time checking of vector instruction support are kind of mutually exclusive, so I would like to see manually (optimized by hand) vectorized functions along with run-time checking.

Reply:
Sure, the `USE_AUTO_VECTORIZER` option is turned off by default, but users can still enable it, if they know what they are doing.

https://github.com/zlib-ng/zlib-ng/blob/915106745a35a8d888c9a78f996648587687a5f3/cmake/toolchain-riscv.cmake#L6

Reply:
Sure, the `USE_AUTO_VECTORIZER` option is turned off by default, but users can still enable it, if they know what they are doing.

https://github.com/zlib-ng/zlib-ng/blob/915106745a35a8d888c9a78f996648587687a5f3/cmake/toolchain-riscv.cmake#L6